Library/CorrelationIdManager.ts

```diff
@@ -0,0 +1,93 @@
+import https = require('https');
+import http = require('http');
+import url = require('url');
+
+class CorrelationIdManager {
+    public static correlationIdPrefix: "cid-v1:";
+
+    private static pendingLookups: {[key: string]: Function[]} = {};
+    private static completedLookups: {[key: string]: string} = {};

```

Comment:
We have manual methods for instrumenting HTTP calls that add headers, more than just the automatic collection client will use this